test/e2e/e2e.go

```diff
@@ -302,6 +303,24 @@ func setupSuite() {
 		nodeKiller := framework.NewNodeKiller(framework.TestContext.NodeKiller, c, framework.TestContext.Provider)
 		go nodeKiller.Run(framework.TestContext.NodeKiller.NodeKillerStopCh)
 	}
+
+	// Allow anonymous access to `/metrics` in tests
```

Comment:
hmm... is this what we expect anyone accessing kcm metrics via the kube-apiserver proxy to do? this also grants access to apiserver /metrics to all anonymous users

Reply:
>  is this what we expect anyone accessing kcm metrics via the kube-apiserver proxy to do?

Currenly I could not think of a better solution. My original purpose of this RBAC rule is to allow anonymous access to `/metrics` **ONLY** in tests, since it is a bit hard to get tests passed after turning off the insecure port. I figured out this way by learning from `perf-tests`: https://github.com/kubernetes/perf-tests/blob/ca470d00ba5e62f77b8afefc8bcb4d5440420fd4/clusterloader2/pkg/prometheus/prometheus.go#L244

Reply:
I think the point is how sensitive the metrics is. 
If metrics is not sensitive, then it is okay to be allow public access to metrics. 
If metrics is sensitive, is it still legitimate to access kcm metrics via the kube-apiserver proxy(which drops Authorization header and doesn't expect pods to see the credential)? Should we use a dedicated service account here?

Reply:
@deads2k @sttts curious how you get metrics from the controller-manager and scheduler pods. any input here?

Reply:
We do token reviews and authn via normal RBAC.

Reply:
> We do token reviews and authn via normal RBAC.

so you access the pod(s) directly or via service, not via the apiserver proxy

Reply:
@s-urbaniak via service endpoints?

Reply:
hello :wave: :-) for kcm, we usually pick up metrics via services and the following selectors: https://github.com/prometheus-operator/kube-prometheus/blob/1bf43811174355359e5316b52bfb1a0b928550b2/manifests/prometheus-serviceMonitorKubeControllerManager.yaml#L54-L59

Reply:
I don't think exposing all /metrics on all components to all anonymous users is reasonable... if we have to do it for this e2e test to get scheduler/kube-controller-manager metrics, it is likely downstream consumers will have to do the same. I'd like to see this switch to pulling metrics from the pod in an authenticated way (maybe via a command run in a collector pod or something? not sure...) that we can point to as an example of how to do it properly.

Reply:
@liggitt How about we add a command to `agnhost`, say `proxy-metrics`, which would listen on `10257` and `10259` , and grab metrics from the secure port of controller-manager and scheduler respectively using the service account token when users access these ports, then we could use it as a reverse-proxy in tests.

Reply:
since anyone actually using the apiserver proxy to fetch metrics from the controlplane pods would have to make adjustments as well, I think we need an approach we could recommend to production consumers as well, which probably means it should not be in a test-oriented pod?

Reply:
@liggitt All tests passed now. I decided to setup a nginx to expose the metrics in the test.